src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx

```diff
@@ -5,30 +5,39 @@ import { ReactComponent as Timer } from '../../../../assets/icons/ic-timer.svg'
 import DeploymentStatusDetailBreakdown from './DeploymentStatusBreakdown'
 import { DeploymentStatusDetailModalType } from './appDetails.type'
 import moment from 'moment'
-import { Moment12HourFormat } from '../../../../config'
+import { Moment12HourFormat, URLS } from '../../../../config'
+import { useHistory, useParams } from 'react-router-dom'
 
 export default function DeploymentStatusDetailModal({
-    close,
     appName,
     environmentName,
+    streamData,
     deploymentStatusDetailsBreakdownData,
 }: DeploymentStatusDetailModalType) {
+    const history = useHistory()
+    const { appId, envId} = useParams<{appId: string , envId: string}>()
     const appStatusDetailRef = useRef<HTMLDivElement>(null)
+
+    const close = () => {
+        const newUrl = `${URLS.APP}/${appId}/${URLS.APP_DETAILS}/${envId}`
+        history.replace(newUrl)
+    }
+
     const escKeyPressHandler = (evt): void => {
```

Comment:
This function is in common we can reuse that

Reply:
path is different in other cases

Reply:
Talking about `escKeyPressHandler`

Reply:
As of now, we don't have it. We'll be creating custom reusable key handler functions/hooks.

Reply:
However, @milindk8 can we try passing onEscape callback as suggested?